testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/authentication/FlowTest.java

```diff
@@ -275,4 +275,150 @@ public void addExecutionFlow() {
         assertAdminEvents.assertEvent(REALM_NAME, OperationType.CREATE, AdminEventPaths.authAddExecutionFlowPath("parent"), params, ResourceType.AUTH_EXECUTION_FLOW);
     }
 
+    @Test
+    //KEYCLOAK-12741
+    public void editFlowTest() {
+        List<AuthenticationFlowRepresentation> flows;
+
+        //copy one first
+        HashMap<String, String> params = new HashMap<>();
+        params.put("newName", "Copy of browser");
+        Response response = authMgmtResource.copy("browser", params);
+        assertAdminEvents.assertEvent(REALM_NAME, OperationType.CREATE, AdminEventPaths.authCopyFlowPath("browser"), params, ResourceType.AUTH_FLOW);
+        try {
+            Assert.assertEquals("Copy flow", 201, response.getStatus());
+        } finally {
+            response.close();
+        }
+
+        //Set a new unique name. Should succeed
+        flows = authMgmtResource.getFlows();
+        AuthenticationFlowRepresentation testFlow = findFlowByAlias("Copy of browser", flows);
+        testFlow.setAlias("Copy of browser2");
+        response = authMgmtResource.updateFlow(testFlow.getId(), testFlow);
+        flows = authMgmtResource.getFlows();
+        try {
+            Assert.assertEquals("Copy of browser2", findFlowByAlias("Copy of browser2", flows).getAlias());
+            assertAdminEvents.assertEvent(REALM_NAME, OperationType.UPDATE, AdminEventPaths.authEditFlowPath(testFlow.getId()), ResourceType.AUTH_FLOW);
+        } finally {
+            response.close();
+        }
+
+        //Create new flow and edit the old one to have the new ones name
+        AuthenticationFlowRepresentation newFlow = newFlow("New Flow", "Test description", "basic-flow", true, false);
+        createFlow(newFlow);
+        // check that new flow is returned in a children list
+        flows = authMgmtResource.getFlows();
+        AuthenticationFlowRepresentation found = findFlowByAlias("New Flow", flows);
+        try {
```

Comment:
This won't need try/finally block if "response" won't be the return type of the "updateFlow" method as I suggested earlier.

Reply:
Edited